Ansible 2.8.8 -> 2.9.7 to fix python 3.8.2 errors

pawl opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
Upgrades Ansible 2.8.8 -> ~~2.9.2~~ 2.9.7 in requirements.txt to fix issues with running Ansible on Python 3.8.2.

## Motivation and Context
I was having the issue described in: https://github.com/trailofbits/algo/issues/1622

```
ERROR! Unexpected Exception, this is probably a bug: cannot pickle '_io.TextIOWrapper' object
```

When attempting to run ./algo with Python 3.8.2 on MacOS 10.14.

## How Has This Been Tested?
I was able to run ./algo and complete an install successfully with the vultr playbook after this change.

## Types of changes
- Bug fix (non-breaking change which fixes an issue)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the code style of this project.
- [x] All new and existing tests passed.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/trailofbits/algo?pullRequest=1777) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/trailofbits/algo?pullRequest=1777) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/trailofbits/algo?pullRequest=1777) it.</sub>

jackivanov commented:
This is a major version change, we need to test the cloud modules

jackivanov commented:
The latest version is 2.9.7 now, so it makes sense to use it

pawl commented:
I updated the PR to 2.9.7: https://github.com/trailofbits/algo/pull/1777/commits/e74b5c7c4c35acaef2b948ab55125e35f10e89c4

I cleared out my .env, reinstalled requirements.txt with ansible 2.9.7, and was still able to deploy successfully to vultr with python 3.8.2.

Ansible put together a porting guide for 2.9:
https://docs.ansible.com/ansible/latest/porting_guides/porting_guide_2.9.html#improved-gather-facts-support-for-network-devices

It says a few modules this project uses are getting deprecated and renamed (with planned removal in ansible 2.13). I made another branch that does some of that renaming: https://github.com/pawl/algo/compare/patch-1...pawl:ansible_2_9_deprecations?expand=1#diff-55a4db0a5082fdc006ebe81b4c673576R9

Let me know if you want me to add that to this PR.

jackivanov commented:
@pawl thanks for the info, let's leave the deprecations for another PR, but instead, we can fix the precautions and docs about the supported python version
